.github/workflows/elixir.yml

```diff
@@ -0,0 +1,301 @@
+        with:
+          path: |
+            _build/${{ env.MIX_ENV }}/lib/supavisor/native
+          key: ${{ runner.os }}-cargo-${{ env.MIX_ENV }}-${{ hashFiles(format('{0}{1}', github.workspace, '/native/**/Cargo.lock')) }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-${{ env.MIX_ENV }}-
+      - name: Compile
+        run: mix compile
+
+  compile-test:
+    name: Compile project in test env
+    runs-on: u22-arm-runner
+    needs: [deps]
+
+    env:
+      MIX_ENV: test
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Elixir
+        uses: erlef/setup-beam@v1
+        with:
+          otp-version: '25.3.2.7'
+          elixir-version: '1.17'
+      - name: Cache Mix
+        uses: actions/cache@v4
+        with:
+          path: deps
+          key: ${{ runner.os }}-mix-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}
+          restore-keys: |
+            ${{ runner.os }}-mix-
+      - name: Cache Build
+        uses: actions/cache@v4
+        with:
+          path: |
+            _build/${{ env.MIX_ENV }}
+            !_build/${{ env.MIX_ENV }}/lib/supavisor/native
+          key: ${{ runner.os }}-build-test-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.MIX_ENV }}-
+      - name: Cache native
+        uses: actions/cache@v4
+        id: native-cache
+        with:
+          path: |
+            priv/native
+          key: ${{ runner.os }}-build-native-${{ hashFiles(format('{0}{1}', github.workspace, '/native/**/*')) }}
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@v1
+        if: steps.native-cache.outputs.cache-hit != 'true'
+        with:
+          toolchain: stable
+      - name: Cache native deps
+        uses: actions/cache@v4
+        if: steps.native-cache.outputs.cache-hit != 'true'
+        with:
+          path: |
+            _build/${{ env.MIX_ENV }}/lib/supavisor/native
+          key: ${{ runner.os }}-cargo-${{ env.MIX_ENV }}-${{ hashFiles(format('{0}{1}', github.workspace, '/native/**/Cargo.lock')) }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-${{ env.MIX_ENV }}-
+      - name: Compile
+        run: mix compile
+
+  format:
+    name: Formatting checks
+    runs-on: u22-arm-runner
+    needs: [deps]
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Elixir
+        uses: erlef/setup-beam@v1
+        with:
+          otp-version: '25.3.2.7'
+          elixir-version: '1.17'
+      - name: Cache Mix
+        uses: actions/cache@v4
+        with:
+          path: deps
+          key: ${{ runner.os }}-mix-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}
+          restore-keys: |
+            ${{ runner.os }}-mix-
+      - name: Run format check
+        run: mix format --check-formatted
+
+  credo:
+    name: Code style
+    runs-on: u22-arm-runner
+    needs: [compile-dev]
```

Comment:
why do you want to check the dev build and not compile-test?

Reply:
Because Credo runs in `dev` environment there.

Reply:
yeah, but probably we can run it for the test env and save a lot of GA time if we do not create a dev release

Reply:
We still create one for `mix dialyze`. But if you want we can converge these with `MIX_ENV=test` for everything.

Reply:
Also, this PR greatly improve caching, so that build times became less of a problem.

Reply:
> We still create one for mix dialyze. But if you want we can converge these with MIX_ENV=test for everything.
Let's try to do that. Basically, there is no difference between the dev and test environments

Reply:
Now everything is in `test` environment.